### PR TITLE
Removes SPIW guns

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
Content "Removes SPIW Flechette Guns"

#### Purpose of change
Flechette based guns were in development since before 1952's Project SALVO, with AAI corporation having worked on them. The subsequent Project SPIW also failed to deliver a working flechette gun, yielding instead the M203 grenade launcher from AAI corp. The Advanced Combat Rifle program would see AAI submit their work, the result of some 34 years of development by one-ish companies to develop a militarily suitable flechette rifle, and see it fail to come to fruition.

Flechette guns are just not that good, especially for applications where individual projectile accuracy are considered important. While they were capable of comparable wounding potential and middling accuracy ( [12 inch mean radius group at 300 yards](https://youtu.be/qFANjlr4I9Q?t=209) ), working out to an ingame dispersion of ~400, [they had really, really bad fliers](https://youtu.be/qFANjlr4I9Q?t=194) associated with issues surrounding the Sabot. Shot patterns can be seen in a1972 trial evaluating closed vs open bolt performance - https://apps.dtic.mil/sti/pdfs/ADA068215.pdf This was noted [as late as '91 ](https://apps.dtic.mil/sti/pdfs/ADA240295.pdf) (see page 9). I was unable to find complete volumes of the trials reports of the ACR trial, other than volume 1, but I think this excerpt easily demonstrates that these problems were not sufficiently resolved even by the time of the ACR trial. image
 
 Why is this important? We could easily model the inaccuracy of this gun. ~400 dispersion is easily modelled. I mention all of this because I do not think that the SPIW weapon would be adopted or *even attempted* by any company. AAI shot their shot. Military development recently in small arms has been focused on the NGSW programs, which have resulted in the adoption of the 6.8×51mm Common Cartridge and rifles firing it. The 'money' is not located in increasing hit probabilities, and no one attempted any single flechette projectiles for defeating armor  - high velocity penetrators have been what we have been seeing, or updates to other calibers (mk 318, m855A1) with penetrators.
 
Rivtech is a company that wears too many hats, and it strains believability that they'd have more than one breakthrough technology. It is my opinion that caseless ammunition based on the much more successful HK G11 should be their 'thing'.

Oh, they were also *[really loud](https://youtu.be/qFANjlr4I9Q?t=659)*.

#### Describe the solution

Removes 5x50 entries and spawns, migrates them all away to caseless equivalents. I am unsure of how to best handle the mods that use it other than generic guns - suggestions welcome.

Updated GG migration tables. Removed corresponding entries from other mods' blacklists.

#### Describe alternatives you've considered

Modelling the flechette guns as shotguns I guess? Updating their stats to be as bad as they should be?

#### Testing

Loaded game, didn't get errors. Visited the Hub.

Loaded game with AFS. No errors.

#### Additional context
None

### CODE REVIEW

**Comment by andrei8l** [comment]
You mention the ACR but all of your data is for the SPIW model. I cannot find any mentions of similar problems with either the AAI or Steyr flechette guns used in the ACR program. I think the passage you mention from the 1991 report is:
```
A major drawback to fielding an individual weapon system incorporating flechette projectiles has been weapon accuracy.
Recent advances in sabot technology show promise in reaching the accuracy levels required for modern individual weapon systems.
```
That second sentence in particular is relevant here, where "recent" could plausibly refer to the ACR program (though the only source there for recent data is an unpublished private work).

I also don't see the problem with Rivtech "wearing a lot of hats". RL arms companies also wear a lot of hats and claim multiple breakthroughs.

**Comment by tenmillimaster** [comment]
As I said before, I do not have a complete copy of the ACR report, just the excerpt I posted. The hit probabilities are poor. 

The passage from the 1991 report is from a year after the ACR trial ended, and it's part of a medical report. I can easily believe that they wouldn't have an expert opinion on the engineering side of things. Better.sabots have not shown up for 34 years. Why try now? The military brass who thought Flechettes were a good idea have retired, they were not attempted for the NGSW.

**Comment by tenmillimaster** [comment]
> FYI, you're current way of doing this is breaking Aftershock. You may have noticed already, but I saw this PR description and figured it was likely.

I've implemented your suggestions as best I could, moving the items over to AFS. I loaded the game with AFS and spawned the gun in, all seemed OK. I didn't adjust the itemspawns or groups, as I was unsure where to put them.

**Comment by fairyarmadillo** [comment]
The flechette guns in the game, excepting the flechette shells used by shotguns, are all modeled after saboted rounds, which have been proven effective in lab settings. One example is the Steyr ACR - https://en.wikipedia.org/wiki/Steyr_ACR 

None of the 5x50s are modeled after the guns described in your source.

Saboted flechettes never saw widespread adoption IRL, but not because of any failure of ballistic performance. They're perfectly sensible in their current implementation as prototype weapons.

**Comment by tenmillimaster** [comment]
> The premise of this PR appears to be built on the assumption that the Rivtech guns are SPIW, which is incorrect. The flechette guns in the game, excepting the flechette shells used by shotguns, are all modeled after ACR rounds, which have been proven effective in testing. One example is the Steyr ACR - https://en.wikipedia.org/wiki/Steyr_ACR

With regards to flechette accuracy out of the Steyr ACR as  opposed to the AAI ACR, they mention that it is worse  in vol 1 of the ACR program summary (page 19). 

> In Steyr's system, the flechette package was crimped by the plastic case head of the round.
This was simply not as strong as the compression applied by a roll-crimped brass case as in the
AAI system. Accordingly, the flechette package left the mouth of the case at a much earlier point
along the pressure-time curve, with the result that round-to-round accuracy suffered when
compared to the AAI round. As mentioned earlier, flechette accuracy has not been at the same
level as bullets even under the best of circumstances. With this system, flechette dispersion was,
at best, twice as wide as the standard bulleted system and often wider than that. This deficiency
became noticeable at the longer ranges. 


It doesn't matter which gun they're modelled after - flechettes inherently have accuracy issues. In Volume 1 of the  ACR Program report, page 22: 

>A third deficiency common to all flechette firing rifles is their lack of accuracy relative to
bulleted approaches. As stated earlier, the accuracy performance of the weapon at this time
is about 1.0 mils linear standard deviation. This contrasts to bulleted weapons on the order of
0.25 mils. The performance improvements projected by Steyr for high precision flechettes and
a muzzle device to eliminate muzzle effects from gas back- flow are now only projections. 

Page 10, same source:
> Looking at performance, the flechette round was not ever made to be as accurate as a standard bulleted
round. There is some controversy as to whether the flechette round is inherently less accurate
and whether any amount of future development effort would result in equal accuracy to a
standard bulleted round. The pace of technology today is such that it is unlikely that further
development work on small caliber flechettes will be funded for rifles. However, flechettes still
offer significant potential for bursting munitions and crew served weapons.

**Comment by andrei8l** [comment]
For reference, there's a copy of ACR report Volume 1 [here](https://www.docdroid.net/ffL1XVy/s00440-pdf). In Appendix B, the accuracy issues for the AAI and Steyr are blamed on quality control for the casing and flechettes, respectively.

Another point is that you're comparing the guns to the M16 like in the ACR program, but the flechette guns here are a SMG and a pistol. In figure 6.3 on page 35, you can see that short range accuracy was comparable to that of the other weapons trialed.

**Comment by fairyarmadillo** [comment]
You said:
> It doesn't matter which gun they're modelled after - flechettes inherently have accuracy issues.

But your own source said:
>  There is some controversy as to whether the flechette round is inherently less accurate
> and whether any amount of future development effort would result in equal accuracy to a
> standard bulleted round. 

The rifles in the trial performed worse than their bulleted counterparts, but the report states that it's not clear whether this is an inherent limitation of flechettes or not. They even account for why Steyr's ACR underperformed, and as andrei8l points out, it's due to design and quality control issues, which a company like Rivtech could easily have solved in the intervening 34 years.

It seems to me like you may have started this PR under the assumption that these were SPIW-style weapons, which would have solid justification for being removed, and are unwilling to change your position even after your initial assumption was shown to be incorrect. Rivtech is a bleeding edge arms manufacturer competing with personal laser weapons, an honest-to-God AI, power armor, and teleportation technology. Spending ungodly amounts of money to correct design flaws that the real-life ACR report suggests _are_ solvable seems perfectly in line with the military industrial complex in our setting.

**Comment by tenmillimaster** [comment]
If I cannot convince you that flechettes are probably inherently inaccurate, then perhaps I can convince you of my other original assertion, that flechettes would not be attempted. 

On January 19, 2017, the XM17 Modular Handgun System concluded, yielding M17's and M18's for the Army and Air Force.
In late 2018, the Army purchased COTS solutions -skipping lengthy R&D and trials time- for Sig MPX SMG's, MCX  Carbines,  716G2 rifles, and SP 2022  pistols.
On March 29, 2019, it was announced the B+T APC9 would fulfill personal security detail armaments as the SCW.
On  April 19, 2022, the Next Generation Squad Weapon Program concluded, yielding M5 rifles and M250 LMG's for the Army.

If the Flechette guns are pistols, why  would they get developed, who would buy them? The military recently bought new guns, they won't be looking for any in some time, and the marines and the navy aren't going to care about the latest, greatest whizbang thing.

If they're SMG's, why would they get developed? The SMG contracts were small potatoes - some 1000 subguns for the MPX, even less, 350, for the APC9.

If they're rifles or MG's of some flavor, why would they get developed? The NGSW just completed and procurement for the m5 and m250 has begun.

Why waste money developing a tech that many in the military doubted, and wouldn't be interested in purchasing for some decades anyways? You could say that Rivtech just dumped money out of its own pocket to develop a product no one would buy, but that's a really big stretch. Or that it tried to develop this tech for one of these trials, but again, even AAI (now Textron systems) didn't try - Textron had their own more conventional entrant into the NGSW program.

**Comment by andrei8l** [comment]
> Why waste money developing a tech that many in the military doubted, and wouldn't be interested in purchasing for some decades anyways?

Because it's cool, same reason Magpul IRL developed the [pocket SMG](https://www.youtube.com/watch?v=enzlTEysYVU) or the [PDR](https://en.wikipedia.org/wiki/Magpul_PDR).

These flechette guns are a cool rare easter egg that go well with the [Protective Lenses and Fingertip Razors CBMs](https://williamgibson.fandom.com/wiki/Molly_Millions). They are at least plausible and not particularly powerful so I don't see the impetus to chop them.

**Comment by fairyarmadillo** [comment]
It's a crowded marketplace. Rivtech struck out on alternative tech that showed some promise in order to differentiate themselves. Per lore, there are ridiculous amounts of money being poured into military R&D due to recent events and all kinds of questionable tech is being explored. The Rivtech guns are prototypes and did not, and perhaps would never have won the contracts they wanted, but they were still developed and put into the testing phase, thus their limited spawns in labs and certain military locations.

**Comment by tenmillimaster** [comment]
When I made changes to the other mods, the maintainers got automatically requested for review, I think? Does this need their review?

File: data/json/items/migration.json [hunk 1]
```diff
@@ -1823,5 +1823,45 @@
     "type": "MIGRATION",
     "replace": "folded_inflatable_boat",
     "reset_item_vars": true
+  },
+  {
+    "id": "5x50dart",
```
Comment (Maleclypse) [inline data/json/items/migration.json:1828]: OK so if this migration exists here then we need to rename the id's on anything in Aftershock that has the same id as something being migrated.   I'd add an `afs_` prefix to the front of the id.

File: data/mods/Aftershock/items/ammo_type.json [hunk 2]
```diff
@@ -52,5 +52,11 @@
     "id": "afs_7.50mm",
     "name": "7.50mm",
     "default": "afs_7.50mm_caseless"
+  },
+  {
+    "type": "ammunition_type",
+    "id": "5x50",
```
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo_type.json:58]:
```suggestion
    "id": "afs_5x50",
```

File: data/mods/Aftershock/items/casing.json [hunk 3]
```diff
@@ -11,5 +11,17 @@
     "symbol": "=",
     "stackable": true,
     "color": "dark_gray"
+  },
+  {
+    "id": "5x50_hull",
```
Comment (Maleclypse) [inline data/mods/Aftershock/items/casing.json:16]:
```suggestion
    "id": "afs_5x50_hull",
```

File: data/mods/Aftershock/items/gun/5x50.json [hunk 4]
```diff
@@ -13,5 +13,84 @@
     "clip_size": 4,
     "flags": [ "NEVER_JAMS", "RELOAD_ONE", "RELOAD_EJECT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "5x50": 4 } } ]
+  },
+  {
+    "id": "needlegun",
```
Comment (Maleclypse) [inline data/mods/Aftershock/items/gun/5x50.json:18]:
```suggestion
    "id": "afs_needlegun",
```

File: data/mods/Aftershock/items/gun/5x50.json [hunk 5]
```diff
@@ -13,5 +13,84 @@
...
+      [ "sling", 1 ],
+      [ "stock", 1 ],
+      [ "stock accessory", 2 ],
+      [ "underbarrel", 1 ]
+    ],
+    "flags": [ "WATERPROOF_GUN", "NEVER_JAMS" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "5x50_100_mag", "5x50_50_mag" ] } ]
+  },
+  {
+    "id": "needlepistol",
```
Comment (Maleclypse) [inline data/mods/Aftershock/items/gun/5x50.json:60]:
```suggestion
    "id": "afs_needlepistol",
```

**Review by Maleclypse — COMMENTED** [review]
I started on it.  I'm not sure if there are things in the 5x50 magazine file that need new ids to prevent them from migrating

**Comment by anothersimulacrum** [comment]
Looks like a bad rebase.

**Comment by tenmillimaster** [comment]
> Looks like a bad rebase.

ughhhhh.

**Comment by tenmillimaster** [comment]
OK I've tried to undo the damage from the botched rebase. I've done it all again and force pushed things. Since there was no way to migrate the needle pistol without some sort of error (holster too small, pistol won't accept magazine, or magazine can't fit # of charges) I've opted for the less aggressive strategy of obsoleting the the needle pistol, needle gun, needle ammo and ammo type. removals where appropriate have happened.

Aftershock now shares duplicate entries with these, prefixed with AFS_, and also a migration file so players don't end up with 2 sets of needle guns.

I've tested the obsoletion/migration in AFS and CDDA with save files that had the relevant items. I also let a needlegun equipped robot shoot at me a bunch in AFS. And I loaded up the Xedra Evolved Scenario that previously had a needlegun.

File: data/mods/Aftershock/recipes/ammo_recipes.json [hunk 6]
```diff
@@ -72,5 +72,29 @@
     "time": "100 m",
     "autolearn": true,
     "using": [ [ "blacksmithing_standard", 16 ], [ "steel_standard", 4 ] ]
+  },
+  {
+    "type": "recipe",
+    "activity_level": "BRISK_EXERCISE",
```
Comment (RenechCDDA) [inline data/mods/Aftershock/recipes/ammo_recipes.json:78]: This is a change from the vanilla recipe, is there any reason why this handloading is moving up *two* activity levels, making it more tiring than any other handloading recipe?

Comment (tenmillimaster) [reply data/mods/Aftershock/recipes/ammo_recipes.json:78]: Fallout from my terrible go at rebasing the first time around. I'll get that fixed.

**Comment by tenmillimaster** [comment]
I'm not sure how to proceed. The tests are failing on generic guns content that, as far as I can tell, should have been broken before I started messing with things. The rm228 migrated to a machine pistol, and the ammo, 20x66 was migrated to shotgun shells.
image
I was getting the above load error before the last commit.

I have no more load errors but the tests failing indicate other issues - I suspect the rm2000 migration is having the same issue - trying to load rifle ammo into a pistol SMG magazine.

**Comment by mqrause** [comment]
The error comes from your changes to Xedra professions, so you need to load Xedra and GG to reproduce. `8mm_hvp` is migrated to `rifle_AP` and `8x40_25_mag` to `rifle_mag` while `rm2000_smg` is migrated to `pistol_smg` and thus can't hold the migrated ammo/mag.

**Comment by Mister-Cultist** [comment]
another in a long line of people removing perfectly functional content for realism

**Comment by RenechCDDA** [comment]
I'm not sure why the impetus was to remove them given the *wholly* negative reaction and the lack of reasoning, either. The argument for removing them was based on misidentifying them in the data and not liking their (fictional!) manufacturer, something that was entirely solvable by modifying their description.

And now that they've been removed there's no way anyone can pass a PR for the supposed alternative of "making their stats as bad as they should be" (even though the data largely supports their previously implemented stats...). I had wrongly assumed that since the reasoning was still proven false this wouldn't be merged even though it was getting new changes, but it seems the comments didn't stop it from being merged regardless of its problems.

**Comment by fairyarmadillo** [comment]
Yeah the justification provided for this PR was provably false and the contributor just sort of ignored that and rammed it through anyway.

**Comment by Zireael07** [comment]
@fairyarmadillo You're referring to their assumption that Rivtech == SPIW?

**Comment by fairyarmadillo** [comment]
@Zireael07 They assumed the rivtech guns were SPIW, yes. When someone pointed out that they were closer to ACR, they tried to apply their same data from the SPIW report to ACR, and were in many places shown to be incorrect, if you scroll up and read through the conversation.

**Pull request merged**